aplib.net-demo/Assets/Scripts/Configuration.cs

```diff
@@ -0,0 +1,277 @@
+    [SerializeField] private const float _minPercentage = 0.2f;
+
+    [SerializeField] private SpawnableEnemies _spawnableEnemies;
+    [SerializeField] private SpawnableItems _spawnableItems;
+    [SerializeField] private GridConfig _gridConfig;
+
+    // Grid component variables.
+    public int MinGridSize { get { return 3; } }
+    public int MaxGridSize { get { return 100; } }
+
+    public int GridVolume { get; private set; }
+    public int GridSizeX { get; private set; }
+    public int GridSizeZ { get; private set; }
+
+    // Room component.
+    public int RoomAmount { get; private set; }
+
+    // Item ammo.
+    public int ItemAmmoAmount { get; private set; }
+
+    // Item health.
+    public int ItemHealthAmount { get; private set; }
+
+    // Item rage.
+    public int ItemRageAmount { get; private set; }
+
+    // Enemy melee.
+    public int EnemyMeleeAmount { get; private set; }
+
+    // Enemy ranged.
+    public int EnemyRangedAmount { get; private set; }
+
+    // Total items.
+    public int TotalItemCount { get; private set; }
+    // Total enemies.
+    public int TotalEnemyCount { get; private set; }
+
+    // Enemy component variables.
+    public bool TestingSwitch { get { return _test; } set { _test = value; } }
+    private bool _test = false;
+
+    // Input fields.
+    public TMP_InputField inputFieldGridW;
+    public TMP_InputField inputFieldGridH;
+    public TMP_InputField inputFieldRoom;
+    public TMP_InputField inputFieldItemAmmo;
+    public TMP_InputField inputFieldItemHealth;
+    public TMP_InputField inputFieldItemRage;
+    public TMP_InputField inputFieldEnemyMelee;
+    public TMP_InputField inputFieldEnemyRanged;
+
+    private void Start()
+    {
+        // Initialize values.
+        GridSizeX = 3;
+        GridSizeZ = 3;
+        GridVolume = GridSizeZ * GridSizeX;
+        RoomAmount = 0;
+        ItemAmmoAmount = 0;
+        ItemHealthAmount = 0;
+        ItemRageAmount = 0;
+        EnemyMeleeAmount = 0;
+        EnemyRangedAmount = 0;
+
+        TotalItemCount = ItemAmmoAmount + ItemHealthAmount + ItemRageAmount;
+        TotalEnemyCount = EnemyRangedAmount + EnemyMeleeAmount;
+    }
+
+    /// <summary>
+    /// Check if input does not exceed the max or min boundry.
+    /// Set valid input.
+    /// </summary>
+    /// <param name="value">Input given from the player through the input field</param>
+    public void SetGridX(string value)
+    {
+        if (value == "") GridSizeX = MinGridSize;
+        else GridSizeX = int.Parse(value);
+        CheckAllValuesToBeValid();
+    }
+
+    /// <summary>
+    /// Check if input does not exceed the max or min boundry.
+    /// Set valid input.
+    /// </summary>
+    /// <param name="value">Input given from the player through the input field</param>
+    public void SetGridZ(string value)
+    {
+        if (value == "") GridSizeZ = MinGridSize;
+        else GridSizeZ = int.Parse(value);
```

Comment:
You can use ternary (?) operator here
```suggestion
        GridSizeZ = value == "" ? MinGridSize : int.Parse(value);
```

Reply:
Changed all that fit the format to this in the script.